src/licenses/LicenseModel.cpp

```diff
@@ -0,0 +1,184 @@
+#include "LicenseModel.h"
+
+#include <QDirIterator>
+
+static const char *const platformRootPath = ":/licenses";
+static const char *const qtRootPath = ":/licenses/qt";
+static const char *const toolkitRootPath = ":/licenses/qt/licenses";
+static const char *const webviewRootPath = ":/licenses/qt/licenses/qtwebengine/src/3rdparty/chromium";
+
+namespace licenses {
+
+  LicenseModel::LicenseModel(QObject *parent) : QAbstractItemModel(parent), m_dir(platformRootPath) {
+    populate();
+  }
+
+  LicenseModel::~LicenseModel() = default;
+
+  void LicenseModel::populate() {
+    QList<QVariant> root_categories;
+    QVector<QVariant> rootData = {QString("Licenses"), QString(platformRootPath)};
+    rootItem = std::make_unique<LicenseItem>(rootData, root_categories);
+
+    QHash<QString, LicenseItem *> map;
+    map[rootItem->path()] = rootItem.get();
+
+    QDirIterator it(m_dir, QDirIterator::Subdirectories);
+
+    while (it.hasNext()) {
+      QString path = it.next();
+
+      QList<QVariant> categories;
+      categories.append(QVariant(static_cast<int>(LicenseCategory::All)));
+
+      if (path.startsWith(webviewRootPath))
+        categories.append(QVariant(static_cast<int>(LicenseCategory::WebView)));
+
+      if (path.startsWith(toolkitRootPath) && !path.startsWith(webviewRootPath))
```

Comment:
why not else-if?

Reply:
Lol, no reason, I was just a bit confused at the time